src/view/directives/autosize.js

```diff
@@ -0,0 +1,76 @@
+import Directive from './directive';
+import {AUTOSIZE_NAME} from '@grid/view/definition';
+
+class Autosize extends Directive(AUTOSIZE_NAME) {
+	constructor($scope, $element, $attrs, $timeout, $document) {
+		super();
+
+		this.$scope = $scope;
+		this.$element = $element;
+		this.$timeout = $timeout;
+		this.$document = $document;
+		this.$attrs = $attrs;
+	}
+
+	onInit() {
+		let $element = this.$element,
+			document = this.$document[0],
+			body = document.body,
+			selector = this.$attrs.autosize,
+			focusElement = null,
+			actualWidth = 0,
+			actualText = '';
+
+		function calculateWidth(text) {
```

Comment:
Can we put this method to core view for autosize to reuse it in the future for other frameworks? Like position.view for position directive.

Reply:
Well, basically we can, but right now it's used only in expression-builder, which is tightly coupled to angularjs. I think we can do it in case it is used within the grid.

Reply:
Yeah, I got. Probably now this refactoring is redundant.